### PR TITLE
Added ACL and comments to Newsrooms

### PR DESCRIPTION
**Pull request by ritave**
Additionally those changes were rebased on top of Registry PR.

### CODE REVIEW

File: src/contracts/newsroom.ts [hunk 1]
```diff
@@ -9,28 +9,134 @@ import { InMemoryProvider } from "../content/providers/inmemoryprovider";
...
+  }
+
+  /**
+   * Checks if the user can assign roles and approve/deny content
+   * Also returns true if user has director super powers
+   * @param address Address for the role check, leave empty for current user
+   * @throws {CivilErrors.NoUnlockedAccount} Requires the node to have at least one account if no address provided
+   */
+  public async isEditor(address?: EthAddress): Promise<boolean> {
+    if (await this.isDirector(address)) {
```
Comment (nickreynolds) [inline src/contracts/newsroom.ts]: should this be below the next block that sets address to current user if not passed in?

File: src/contracts/newsroom.ts [hunk 2]
```diff
@@ -9,28 +9,134 @@ import { InMemoryProvider } from "../content/providers/inmemoryprovider";
...
+  }
+
+  /**
+   * Checks if the user can propose content to the Newsroom
+   * Also returns true if user has director super powers
+   * @param address Address for the role check, leave empty for current user
+   * @throws {CivilErrors.NoUnlockedAccount} Requires the node to have at least one account if no address provided
+   */
+  public async isReporter(address?: EthAddress): Promise<boolean> {
+    if (await this.isDirector(address)) {
```
Comment (nickreynolds) [inline src/contracts/newsroom.ts]: same as above, should this be below the part where address is set?

File: src/contracts/newsroom.ts [hunk 3]
```diff
@@ -93,4 +216,29 @@ export class Newsroom extends BaseWrapper<NewsroomContract> {
...
+
+// TODO(ritave): generate roles from smart-contract
+/**
+ * Roles that are supported by the Newsroom
+ * - Director has full access to everything
+ * - Editor can approve or deny contant, as well as assigning roles to actors
+ * - Reported who can propose content for the Editors to approve
+ */
+export enum Roles {
+  Director = "director",
```
Comment (nickreynolds) [inline src/contracts/newsroom.ts]: I'm in favor of ALL_CAPS_WITH_UNDERSCORES for enum elements. Do you feel otherwise?

File: src/utils/errors.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Web3Wrapper } from "./web3wrapper";
+
+export enum CivilErrors {
+  NoUnlockedAccount = "NO_UNLOCKED_ETHEREUM_ACCOUNT",
+  NoPrivileges = "NOT_ENOUGH_PRIVILEGES_FOR_ACTION",
```
Comment (nickreynolds) [inline src/utils/errors.ts]: same as other enum, NoPrivileges -> NO_PRIVILEGES, but I'm open to this convention if you feel strongly

File: src/civil.ts [hunk 5]
```diff
@@ -36,13 +49,42 @@ export class Civil {
     this.abiDecoder = new AbiDecoder(Object.values<Artifact>(artifacts).map((a) => a.abi));
   }
 
+  /**
+   * @returns Currently default user account used, undefined if none unlocked/found
+   */
+  public get account() {
```
[the remark this answers is not in the document]
Comment (ritave) [reply src/civil.ts]: Done

File: src/contracts/newsroom.ts (hunk 1 above)
Comment (ritave) [reply src/contracts/newsroom.ts]: No, it's checked in `isDirector` anyways

File: src/contracts/newsroom.ts (hunk 2 above)
Comment (ritave) [reply src/contracts/newsroom.ts]: No, it's checked in `isDirector` anyways

File: src/contracts/newsroom.ts (hunk 3 above)
Comment (ritave) [reply src/contracts/newsroom.ts]: I have no preference at all, checked Typescript documentation -> They're doing the Capitalized variant, https://www.typescriptlang.org/docs/handbook/enums.html

Might as well stay with it

File: src/utils/errors.ts (hunk 4 above)
Comment (ritave) [reply src/utils/errors.ts]: I have no preference at all, checked Typescript documentation -> They're doing the Capitalized variant, https://www.typescriptlang.org/docs/handbook/enums.html

Might as well stay with it

**Pull request merged**